### PR TITLE
Add junit mocha reporter

### PR DESCRIPTION
**Pull request by srknzl**
With this we will be able to see running test and test results in jenkins.

### CODE REVIEW

File: test/integration/backward_compatible/LoggingTest.js [hunk 1]
```diff
@@ -69,15 +71,15 @@ describe('LoggingTest', function () {
...
+        consoleLogSpy = sinon.spy(console, 'log');
     });
 
     afterEach(async function () {
         if (client != null) {
             await client.shutdown();
             client = null;
         }
-        console.log.restore();
+        consoleLogSpy.restore();
```
Comment (srknzl) [inline test/integration/backward_compatible/LoggingTest.js:82]: this reporter changes how console.log works, so we need to use variable here for this test to pass

File: package.json [hunk 2]
```diff
@@ -21,6 +21,7 @@
         "eslint-plugin-mocha": "~8.1.0",
         "jsonschema": "~1.4.0",
         "mocha": "~8.3.2",
+        "mocha-jenkins-reporter": "^0.4.5",
```
Comment (mdumandag) [inline package.json:24]: Do we still need `mocha-junit-reporter": "~2.0.0",` dependency?

File: .mocharc.json [hunk 3]
```diff
@@ -4,5 +4,11 @@
     "exclude": [
         "test/soak_test/**",
         "test/user_code/**"
+    ],
+    "reporter": "mocha-jenkins-reporter",
+    "reporter-option": [
+        "junit_report_name=Node.js Client Tests",
+        "junit_report_path=report.xml",
+        "junit_report_stack=1"
```
Comment (mdumandag) [inline .mocharc.json:12]: Will this affect the stack we see on test failures? It would be very helpful if we can see the full stack trace on test failures

Comment (srknzl) [reply .mocharc.json:12]: not it enables stack, we can see trace

File: package.json (hunk 2 above)
Comment (srknzl) [reply package.json:24]: guess not

**Comment by srknzl** [comment]
verify

**Review by mdumandag — APPROVED** [review]

**Pull request merged**